src/Xenial.Tasty/Execution/TestRuntime/RunCommandMiddleware.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace Xenial.Delicious.Execution.TestRuntime
+{
+    public static class RunCommandMiddleware
+    {
+        public static TestExecutor UseRunCommands(this TestExecutor executor)
+            => executor.UseRuntime(async (context, next) =>
+            {
+                try
+                {
+                    if (context.CurrentCommand != null)
+                    {
+                        await context.CurrentCommand.Command(context);
```

Comment:
Error reporting? We can use a generic error catching middleware I guess. Not sure if we should report this to test reporters though

Reply:
This should be covered by closer look at it